### PR TITLE
feat(elasticache): Ensure Redis Cache Clusters Automatically Install Minor Updates

### PR DESCRIPTION
**Pull request by HugoPBrito**
### Context

To enhance Prowler's functionality, we have implemented a new check that evaluates whether ElastiCache for Redis automatically applies minor version upgrades to cache clusters by assessing the `AutoMinorVersionUpgrade` parameter.

### Description

I have added a new check called `elasticache_redis_cluster_auto_minor_version_upgrades` along with its respective tests to verify its functionality.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        findings = []
+        for cluster in elasticache_client.clusters.values():
+            if cluster.engine == "redis":
+                report = Check_Report_AWS(self.metadata())
+                report.region = cluster.region
+                report.resource_id = cluster.id
+                report.resource_arn = cluster.arn
+                report.resource_tags = cluster.tags
+                report.status = "PASS"
+                report.status_extended = f"Elasticache Redis cache cluster {cluster.id} does have automated minor version upgrades enabled."
```
Comment (sergargar) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py:18]:
```suggestion
                report.status_extended = f"Elasticache Redis cluster {cluster.id} does have automated minor version upgrades enabled."
```

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+                if not getattr(cluster, "auto_minor_version_upgrade", False):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py]:
```suggestion
                if not cluster.auto_minor_version_upgrade:
```
You are setting a default value in the service so the `getattr` is not needed.

File: prowler/providers/aws/services/elasticache/elasticache_service.py [hunk 3]
```diff
@@ -158,6 +160,7 @@ class Cluster(BaseModel):
     cache_subnet_group_id: Optional[str]
     subnets: list = []
     tags: Optional[list]
+    auto_minor_version_upgrade: Optional[bool] = False
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_service.py]:
```suggestion
    auto_minor_version_upgrade: bool = False
```

Comment (MrCloudSec) [reply prowler/providers/aws/services/elasticache/elasticache_service.py]: You are using the attribute when initializing the class, so it is not optional.

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.elasticache.elasticache_client import (
+    elasticache_client,
+)
+
+
+class elasticache_redis_cluster_auto_minor_version_upgrades(Check):
+    def execute(self):
+        findings = []
+        for cluster in elasticache_client.clusters.values():
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py]: Please, test this in a real environment, I think we should use the `replication_groups` as in the other checks.

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "high",
+  "ResourceType": "AWSElastiCacheClusters",
+  "Description": "Ensure Elasticache Redis cache clusters have automatic minor upgrades enabled.",
+  "Risk": "Not enabling automatic minor version upgrades can expose your Redis cluster to security vulnerabilities, performance issues, and increased operational overhead due to the need for manual updates.",
+  "RelatedUrl": "https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/VersionManagement.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "aws elasticache modify-cache-cluster --cache-cluster-id <cluster_id> --apply-immediately --auto-minor-version-upgrade",
+      "NativeIaC": "",
+      "Other": "",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.metadata.json]:
```suggestion
      "Other": "https://docs.aws.amazon.com/securityhub/latest/userguide/elasticache-controls.html#elasticache-2",
```

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.metadata.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure Elasticache clusters have automatic minor upgrades enabled.",
+      "Url": "https://docs.aws.amazon.com/securityhub/latest/userguide/elasticache-controls.html#elasticache-2"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.metadata.json]: Find another related URL please.

File: prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py [hunk 7]
```diff
@@ -7,22 +7,21 @@
...
+            report.resource_arn = repl_group.arn
+            report.resource_tags = repl_group.tags
+            report.status = "PASS"
+            report.status_extended = f"Elasticache Redis cache cluster {repl_group.id} does have automated minor version upgrades enabled."
 
+            if not repl_group.auto_minor_version_upgrade:
+                report.status = "FAIL"
+                report.status_extended = f"Elasticache Redis cache cluster {repl_group.id} does not have automated minor version upgrades enabled."
                 findings.append(report)
+                break
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades.py]: Why this break?

File: tests/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades_test.py [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+)
+from tests.providers.aws.utils import AWS_REGION_US_EAST_1, set_mocked_aws_provider
+
+VPC_ID = "vpc-12345678901234567"
+
+
+# Patch every AWS call using Boto3
+@patch("botocore.client.BaseClient._make_api_call", new=mock_make_api_call)
+class Test_elasticache_redis_cluster_auto_minor_version_upgrades:
+    @mock_aws
```
Comment (MrCloudSec) [inline tests/providers/aws/services/elasticache/elasticache_redis_cluster_auto_minor_version_upgrades/elasticache_redis_cluster_auto_minor_version_upgrades_test.py]: Why do you use moto if you are not creating the resources with boto3?

**Review by MrCloudSec — APPROVED** [review]
👏🏼 👏🏼

**Pull request merged**